pyscraper/new_hansard.py

```diff
@@ -0,0 +1,1442 @@
+        elif tag_name == 'hs_CLPara':
+            self.parse_witness(tag)
+        elif tag_name == 'hs_2BillTitle':
+            self.parse_bill_title(tag)
+        elif tag_name == 'hs_3MainHdg':
+            self.committee_finished()
+            self.parse_major(tag)
+        elif tag_name == 'hs_76fChair':
+            self.parse_chair(tag)
+        elif tag_name == 'hs_ParaIndent':
+            self.parse_para_with_member(tag, None, css_class="indent")
+        else:
+            handled = super(PBCParseDayXML, self).handle_tag(tag_name, tag)
+
+        return handled
+
+
+class LordsParseDayXML(BaseParseDayXML):
+    resolver = LordsPimsList()
+
+    paras = [
+        'hs_para',
+        'hs_quote',
+        'hs_quotefo',
+        'hs_parafo',
+        'hs_Question',
+        'hs_newline10',
+    ]
+
+    ignored_tags = [
+        'hs_date',
+        'hs_Venue',
+    ]
+
+    def parse_quote(self, quote):
+        tag = etree.Element('p')
+        tag.set('pid', self.get_pid())
+        tag.set('class', 'indent')
+
+        tag.text = re.sub('\n', ' ', quote.text)
+
+        i = quote.xpath('./ns:I', namespaces=self.ns_map)
+        if len(i) == 1:
+            i_text = self.get_single_line_text_from_element(i[0])
+            new_i = etree.Element('i')
+            new_i.text = i_text
+            new_i.tail = re.sub('\n', ' ', i[0].tail)
+            if re.match(r'Official Report,?$', i_text):
+                phrase = etree.Element('phrase')
+                phrase.set('class', 'offrep')
+                # FIXME: generate a proper id here
+                phrase.set('id', new_i.tail)
+                phrase.append(new_i)
+                tag.append(phrase)
+            else:
+                tag.append(new_i)
+
+        self.current_speech.append(tag)
+
+    def parse_member(self, member):
+        found_member = super(LordsParseDayXML, self).parse_member(member)
+        if found_member is None:
+            if member.get('PimsId') == 0:
+                found_member = {
+                    'person_id': 'unknown',
+                    'name': u''.join(member.xpath('.//text()'))
+                }
+
+        return found_member
+
+    def parse_newdebate(self, tag):
+        time = tag.xpath('.//ns:hs_time', namespaces=self.ns_map)
+        if len(time):
+            self.parse_time(time[0])
+
+        heading = tag.xpath('.//ns:hs_DebateHeading', namespaces=self.ns_map)
+        debate_type = tag.xpath('.//ns:hs_DebateType', namespaces=self.ns_map)
+        if len(heading):
+            if len(debate_type):
+                text = self.get_single_line_text_from_element(debate_type[0])
+                self.parse_major(heading[0], extra_text=text)
+            else:
+                self.parse_major(heading[0])
+        else:
+            sys.stderr.write('newdebate with no heading', namespaces=self.ns_map)
+            return
+
+        #procedure = tag.xpath('.//ns:hs_Procedure', namespaces=self.ns_map)
```

Comment:
I commented this out because there's no "handle_para" function that would have got called. Does this need putting back in in some way?